### PR TITLE
Issue/5997 Variations bulk update sale price

### PR DESCRIPTION
**Pull request by samiuelson**
Note: this PR implements the https://github.com/woocommerce/woocommerce-android/issues/5997 partially and targets the base feature branch. It's a continuation of https://github.com/woocommerce/woocommerce-android/pull/6566 PR and should be merged after that one. Marking it as a draft until then.

### Description
This PR adds support for sale price bulk updating along with a few improvements.
* Allowing to bulk update sale prices.
* Showing progress bar dialog and blocking the UI during the bulk update operation.
* Supporting bulk updating prices to "empty" values

### Testing instructions
1. Open product details and go to "Variations" list (product must have variations defined).
2. After variations are loaded click the "Bulk update..." menu item.
3. The bottom sheet should appear. Choose "Sale price".
4. Enter the new regular price value and click "Done".
5. Price update screen should be closed after successful operation and a confirmation snack should appear. In case of error, the screen should not be closed and the error snack should be shown.

### Visual changes

https://user-images.githubusercontent.com/4527432/170757742-236ae335-05bb-4fed-80db-02f1d35ad4de.mp4

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/variations/VariationsBulkUpdateAttrPickerDialog.kt [hunk 1]
```diff
@@ -87,7 +88,20 @@ class VariationsBulkUpdateAttrPickerDialog : WCBottomSheetDialogFragment() {
...
                 if (price != null && currency != null) {
-                    currencyFormatter.formatCurrency(amount =  price, currencyCode = currency)
+                    currencyFormatter.formatCurrency(amount = price, currencyCode = currency)
+                } else {
+                    ""
+                }
+            }
+        }
+        binding.salePriceSubtitle.text = when (newState.salePriceGroupType) {
+            ValuesGroupType.None -> getString(R.string.variations_bulk_update_dialog_price_none)
```
Comment (0nko) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/variations/VariationsBulkUpdateAttrPickerDialog.kt]: This code seems like a duplicate of the one used for the regular price. WDYT about extracting it to a function?

**Review by 0nko — COMMENTED** [review]
Looks great, @samiuelson! I just left a minor code comment.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/variations/VariationsBulkUpdateAttrPickerDialog.kt (hunk 1 above)
Comment (samiuelson) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/variations/VariationsBulkUpdateAttrPickerDialog.kt]: Good catch, thanks. Fixed in e9d1bea.

**Review by 0nko — APPROVED** [review]
:shipit:

**Pull request merged**